File: 8kyu/(8 kyu) To square or not/(8 kyu) To square or not.go
package kata

import "math"

func SquareOrSquareRoot(arr []int) []int{
  var res []int

  for _, x := range arr {
    squaredNumber := math.Pow(float64(x), .5)

    if squaredNumber == math.Trunc(squaredNumber) {
      res = append(res, int(squaredNumber))
    } else {
      res = append(res, int(math.Pow(float64(x), 2)))
    }
  }

  return res
}

File: 8kyu/(8 kyu) To square or not/(8 kyu) To square or not.py
import math

def square_or_square_root(arr):
    res = []

    for x in arr:
        squared_number = math.sqrt(x)
        if squared_number == math.trunc(squared_number):
            res.append(squared_number)
        else:
            res.append(math.pow(x, 2))

    return res
